src/Core/Managed/Shared/DataContracts/DependencyTelemetry.cs

```diff
@@ -201,6 +214,7 @@ void ITelemetry.Sanitize()
         {
             this.Name = this.Name.SanitizeName();
             this.Name = Utils.PopulateRequiredStringValue(this.Name, "name", typeof(DependencyTelemetry).FullName);
+            this.Id = this.Id.SanitizeName();
```

Comment:
What should the max length for ResultCode be when we sanitize? And do we need to sanitize Id?